### PR TITLE
Changed duplicate detection test call URL

### PR DESCRIPTION
**Pull request by kartikeyrajvaidya**

### CODE REVIEW

File: src/app/core/services/duplicate-detection.service.ts [hunk 1]
```diff
@@ -30,12 +30,8 @@ export class DuplicateDetectionService {
     return ['Different expense', 'Other'];
   }
 
-  get(url, config?) {
-    return this.httpClient.get(this.ROOT_ENDPOINT + '/property_evaluator' + url, config);
-  }
-
   post(url, data, config?) {
```
Comment (Julias0) [inline src/app/core/services/duplicate-detection.service.ts:33]: Add data type in following format - 
post(url: string, data: { param1: string, param2: number }, config: { wth_is_this: is_this_required }): Observable<ComplianceResponse> {

File: src/app/core/services/duplicate-detection.service.ts [hunk 2]
```diff
@@ -64,12 +60,8 @@ export class DuplicateDetectionService {
           transactionCopy.from_dt.setMilliseconds(0);
           transactionCopy.from_dt = this.timezoneService.convertToUtc(transactionCopy.from_dt, localeOffset);
         }
-        return this.post('/duplicate/test', transactionCopy);
+        return this.post('/l1_test', transactionCopy);
```
Comment (Julias0) [inline src/app/core/services/duplicate-detection.service.ts:63]: same as above?

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 3]
```diff
@@ -488,15 +481,52 @@ export class AddEditExpensePage implements OnInit {
...
+        currentExpenseObject: currentExpenseObject$,
+        originalExpenseObject: this.etxn$
+      }).pipe(
+        map(({currentExpenseObject, originalExpenseObject}) => {
+          for (const fieldName of duplicateFieldsToBeCompared) {
+
+            if (['txn_dt', 'to_dt', 'from_dt'].includes(fieldName)) {
+              let currentDate = currentExpenseObject.tx[fieldName] && moment(currentExpenseObject.tx[fieldName]).format('y-MM-DD');
+              let originalDate = originalExpenseObject.tx[fieldName] && moment(originalExpenseObject.tx[fieldName]).format('y-MM-DD');
+              if (!isEqual(currentDate, originalDate)) {
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:500]: `return !isEqual(currentDate, originalDate) ` no need for if condition

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 4]
```diff
@@ -488,15 +481,52 @@ export class AddEditExpensePage implements OnInit {
...
+                return true;
+              }
+            } else {
+              if (!isEqual(currentExpenseObject.tx[fieldName], originalExpenseObject.tx[fieldName])) {
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:504]: same as above

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 5]
```diff
@@ -280,15 +280,52 @@ export class AddEditMileagePage implements OnInit {
...
   }
 
   setupDuplicateDetection() {
     this.duplicates$ = this.fg.valueChanges.pipe(
       debounceTime(1000),
-      distinctUntilChanged((a, b) => isEqual(a, b)),
-      switchMap(() => {
-        return this.getPossibleDuplicates();
+      switchMap(() => this.getPossibleDuplicates()),
+      switchMap((test) => {
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:323]: test -> isPossibleDuplicate

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 6]
```diff
@@ -488,15 +481,52 @@ export class AddEditExpensePage implements OnInit {
   }
 
   getPossibleDuplicates() {
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:483]: I see duplication in expense, mileage and per diem - can you look into reducing it?

**Review by Julias0 — CHANGES_REQUESTED** [review]
Lets take care of duplication too

**Review by abhishek1234321 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**